spec/unit/backtracie_spec.rb

```diff
@@ -223,6 +223,30 @@ class ClassWithMethodDefinedUsingDefinedMethod
       end
     end
 
+    context "when sampling a singleton method defined using define_method" do
+      class SingletonClassWithMethodDefinedUsingDefinedMethod
+        class << self
+          define_method(:test_method) do |&block|
+            block.call
+          end
+        end
+      end
+
+      let(:test_object) { SingletonClassWithMethodDefinedUsingDefinedMethod }
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_object.test_method { described_class.backtrace_locations(Thread.current) }, test_object.test_method { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it "sets the qualified_method_name to include the class name and test_method{block}" do
```

Comment:
Good point that this is one more case where we get stuck with that annoying `{block}` suffix. Perhaps something to try asking ruby-core to change in the future?

Reply:
I was thinking in the shower about whether it's possible to define a bmethod that _isn't_ a block. Like.... what happens if you do

```
def foo; end;
define_method :bar, &method(:foo)
```

I'll have to try it :)